packages/dds/tree/src/simple-tree/discrepancies.ts

```diff
@@ -294,21 +443,42 @@ function* getFieldDiscrepancies(
 		} satisfies AllowedTypeDiscrepancy;
 	}
 
-	if (view.kind !== stored.kind) {
+	const viewKind =
+		convertFieldKind.get(view.kind) ??
+		fail("A conversion from a FieldKind to a FlexFieldKind should exist");
+
+	// This checks if the field kind in the view schema is not compatible with the stored schema.
+	// We cannot detect if the view schema is a sequence using the kind property so it is passed in separately.
+	// TODO: This is a temporary workaround until the comparison logic is redesigned.
```

Comment:
I think this TODO should be move to the viewKindIsSequence parameter of this function, as its currently unclear that having it is a temporary workaround unless you read the implementation all the way to here.